sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1429,3 +1293,95 @@ class SQLContext(@transient val sparkContext: SparkContext)
 }
 
 
+
+/**
+ * :: DeveloperApi ::
+ * The primary workflow for executing relational queries using Spark.  Designed to allow easy
+ * access to the intermediate phases of query execution for developers.
+ */
+@DeveloperApi
+protected[sql] class QueryExecution(val sqlContext: SQLContext, val logical: LogicalPlan) {
```

Comment:
It is `protected[sql]` in master. 

BTW, this PR is geared towards making it easier for third parties (and HiveContext) to add new processing rules without requiring to subclass (see PR #5556 and [SPARK-6320](https://issues.apache.org/jira/browse/SPARK-6320)) 

I would actually advise making these classes *public*